core/src/main/java/io/questdb/griffin/SqlOptimiser.java

```diff
@@ -4799,6 +4808,57 @@ && isHourKeyword(qc.getAst().token) && qc.getAst().paramCount == 1 && qc.getAst(
         return root;
     }
 
+
+    private boolean aliasAppearsInColsAndColArgs(QueryModel model, CharSequence alias) {
+        boolean appearsInCols = false;
+        boolean appearsInArgs = false;
+        ObjList<QueryColumn> modelColumns = model.getColumns();
+        for (int i = 0; i < modelColumns.size(); i++) {
+            QueryColumn col = modelColumns.get(i);
+            if (col.getAst().type == 8) {
+                // function
+                ExpressionNode rhs = col.getAst().rhs;
+                ExpressionNode argNode = null;
+                while (rhs != null) {
```

Comment:
This loop doesn't seem to handle complex nested cases, e.g. `concat(lpad(str_col1, 10, '0'), lpad(str_col2_alias, 32, '0'))`. Apart from the code changes, some tests should be added.